### ISSUE TITLE
Serialization support

### ISSUE THREAD

**Issue by jneem** [AltSysrq/proptest#116]
tl;dr: Is it possible for proptest to support some sort of configurable formatting and/or serialization instead of relying on Debug?

So proptest found a bug for me (thanks!) but the test case is fairly large and so understanding it is a bit of a pain. Fortunately, the types I'm generating can be serialized with serde into human-readable formats (whereas the default {:?} formatting that proptest uses in proptest-regressions isn't readable for my types).

One option that would be convenient for me would be to support an attribute (say `#[proptest_formatter="some_function"]`) that specifies a function for turning the generated arguments into a `Vec<u8>`. Then when persisting the test-case, proptest could use that function instead of the default Debug formatter. Does that sound reasonable?

**Comment by Centril** [comment]
I think we should use the `log` crate; this is what the `quickcheck` does. With that you can easily customize how you want logging to occur.

**Comment by Eh2406** [comment]
Another workaround is to use a new type with a custom debug impl. [Cargo uses this for its tests](https://github.com/rust-lang/cargo/blob/master/tests/testsuite/support/resolver.rs#L273).
